mod_auth/controllers.py

```diff
@@ -59,7 +59,8 @@ def login_required(f: Callable) -> Callable:
     def decorated_function(*args, **kwargs):
         if g.user is None:
             g.log.warning(f'login protected endpoint {request.endpoint} accessed before logging in')
-            return redirect(url_for('auth.login', next=request.endpoint))
+            # Store the full URL path instead of just the endpoint name to preserve URL parameters
+            return redirect(url_for('auth.login', next=request.full_path.rstrip('?')))
```

Comment:
If I'm not mistaken this could lead to security risks, are you sure this is safe?

Reply:
Addressed - reverted to using endpoint name and added try/except around url_for() to catch BuildError gracefully. Now redirects to home page when the endpoint requires parameters we don't have.